agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3701,90 +3701,97 @@ bool slave_thread::slave_fsm(bool &call_slave_select)
             return false;
         }
 
-        auto notification =
-            message_com::add_vs_tlv<beerocks_message::cACTION_CONTROL_SLAVE_JOINED_NOTIFICATION>(
-                cmdu_tx);
-
-        if (!notification) {
-            LOG(ERROR) << "Failed building cACTION_CONTROL_SLAVE_JOINED_NOTIFICATION!";
-            return false;
-        }
-
-        notification->is_slave_reconf() = is_backhual_reconf;
-        is_backhual_reconf              = false;
-
-        // Version
-        string_utils::copy_string(notification->slave_version(message::VERSION_LENGTH),
-                                  BEEROCKS_VERSION, message::VERSION_LENGTH);
-
-        // Platform Configuration
-        notification->platform()             = config.platform;
-        notification->low_pass_filter_on()   = config.backhaul_wireless_iface_filter_low;
-        notification->enable_repeater_mode() = config.enable_repeater_mode;
-        notification->radio_identifier() = network_utils::mac_from_string(config.radio_identifier);
-
-        // Backhaul Params
-        notification->backhaul_params().gw_ipv4 =
-            network_utils::ipv4_from_string(backhaul_params.gw_ipv4);
-        notification->backhaul_params().gw_bridge_mac =
-            network_utils::mac_from_string(backhaul_params.gw_bridge_mac);
-        notification->backhaul_params().is_backhaul_manager = is_backhaul_manager;
-        notification->backhaul_params().backhaul_iface_type = backhaul_params.backhaul_iface_type;
-        notification->backhaul_params().backhaul_mac =
-            network_utils::mac_from_string(backhaul_params.backhaul_mac);
-        notification->backhaul_params().backhaul_channel = backhaul_params.backhaul_channel;
-        notification->backhaul_params().backhaul_bssid =
-            network_utils::mac_from_string(backhaul_params.backhaul_bssid);
-        notification->backhaul_params().backhaul_is_wireless = backhaul_params.backhaul_is_wireless;
-
-        if (!config.bridge_iface.empty()) {
-            notification->backhaul_params().bridge_mac =
-                network_utils::mac_from_string(backhaul_params.bridge_mac);
-            notification->backhaul_params().bridge_ipv4 =
-                network_utils::ipv4_from_string(backhaul_params.bridge_ipv4);
-            notification->backhaul_params().backhaul_ipv4 =
-                network_utils::ipv4_from_string(backhaul_params.bridge_ipv4);
+        if (config.non_intel) {
```

Comment:
The idea behind supporting Intel controller is that each company can differentiate itself with extra features which are not supported in the standard. For example, DCS and DFS support can be negotiated during autoconfiguration:
The agent includes a VS TLV in the autoconfiguration CMDU which marks its extra capabilities.
prplMesh controller understands that, and adds its own VS TLV to the M2, just like we do today.

Coming to think about it, we have the manufacturer string in the M1 and M2 messages, so instead of using a global flag I suggest to read the manufacturer from the configuration file, and only if its Intel send the VS stuff, in both the controller and the agent.

So the flag you added would just replace the manufacturer in the configuration file from Intel (which should become the default) to something else.

While we're at it - lets make the manufacturer a CMAKE parameter, default being Intel.

Reply:
> For example, DCS and DFS support can be negotiated during autoconfiguration:

DFS is an R2 feature so not a good example :-)

> The agent includes a VS TLV in the autoconfiguration CMDU which marks its extra capabilities.
> prplMesh controller understands that, and adds its own VS TLV to the M2, just like we do today.

I think we'll want to emulate a non-Intel controller as well, i.e. controller would ignore the VS TLV.
Conversely, there is no actual need for the controller to remove the VS TLVs when talking to a non-Intel agent, because the agent should ignore unknown VS TLVs.


> Coming to think about it, we have the manufacturer string in the M1 and M2 messages, so instead of using a global flag I suggest to read the manufacturer from the configuration file, and only if its Intel send the VS stuff, in both the controller and the agent.

As you note elsewhere: integrators might want to change the manufacturer string but still use the extra features.


> So the flag you added would just replace the manufacturer in the configuration file from Intel (which should become the default) to something else.

I think the config file should be removed entirely, and instead bpl (or something similar) should be used to tie in with the system's config store. Well, config file *may* still be useful for debugging/experimentation options, but for those I'd actually prefer them to come through bml.

> While we're at it - lets make the manufacturer a CMAKE parameter, default being Intel.

I think the default should be prpl :-)